plugins/FrequencyShifter/FrequencyShifterEffect.cpp

```diff
@@ -0,0 +1,294 @@
+
+#include "embed.h"
+#include "plugin_export.h"
+
+#include <algorithm>
+#include <cmath>
+#include <numbers>
+
+namespace lmms
+{
+
+extern "C"
+{
+Plugin::Descriptor PLUGIN_EXPORT frequencyshifter_plugin_descriptor =
+{
+	LMMS_STRINGIFY(PLUGIN_NAME),
+	"Frequency Shifter",
+	QT_TRANSLATE_NOOP("PluginBrowser", "A frequency shifter (not a pitch shifter) and barberpole phaser plugin"),
+	"Lost Robot <r94231/at/gmail/dot/com>",
+	0x0100,
+	Plugin::Type::Effect,
+	new PluginPixmapLoader("logo"),
+	nullptr,
+	nullptr,
+};
+}
+
+FrequencyShifterEffect::FrequencyShifterEffect(Model* parent, const Descriptor::SubPluginFeatures::Key* key) :
+	Effect(&frequencyshifter_plugin_descriptor, parent, key),
+	m_controls(this)
+{
+	m_ringBuf.resize(1, {0.f, 0.f});
+
+	connect(Engine::audioEngine(), &AudioEngine::sampleRateChanged,
+		this, &FrequencyShifterEffect::updateSampleRate);
+	updateSampleRate();
+}
+
+Effect::ProcessStatus FrequencyShifterEffect::processImpl(SampleFrame* buf, const fpp_t frames)
+{
+	const float twoPi = std::numbers::pi_v<float> * 2.0f;
+
+	const float mix = m_controls.m_mix.value() * wetLevel();
+	const float fs = m_controls.m_freqShift.value();
+	const float spread = m_controls.m_spreadShift.value();
+	const float ring = m_controls.m_ring.value();
+	const float feedback = m_controls.m_feedback.value();
+	const float delayLen = (m_controls.m_delayLengthLong.value() + m_controls.m_delayLengthShort.value()) * 0.001f * m_sampleRate;
+	const float delayDamp = m_controls.m_delayDamp.value();
+	const float delayGlide = m_controls.m_delayGlide.value();
+	const float lfoAmt = m_controls.m_lfoAmount.value();
+	const float lfoRate = (m_controls.m_lfoRate.value() / m_sampleRate) * twoPi;
+	const float lfoSt = m_controls.m_lfoStereoPhase.value() * twoPi;
+	const bool antireflect = m_controls.m_antireflect.value();
+	const int routeMode = m_controls.m_routeMode.value();
+	const float harmonics = m_controls.m_harmonics.value();
+	const float glide = m_controls.m_glide.value();
+	const float tone = m_controls.m_tone.value();
+	const float phase = m_controls.m_phase.value() * twoPi;
+
+	const bool resetShifterBtn = m_controls.m_resetShifter.value();
+	const bool resetLfoBtn = m_controls.m_resetLfo.value();
+
+	if (!m_prevResetShifter && resetShifterBtn)
+	{
+		m_phase[0] = 0.f;
+		m_phase[1] = 0.f;
+	}
+	if (!m_prevResetLfo && resetLfoBtn)
+	{
+		m_lfoPhase = 0.f;
+	}
+	m_prevResetShifter = resetShifterBtn;
+	m_prevResetLfo = resetLfoBtn;
+
+	const float twoPiOverSr = twoPi / m_sampleRate;
+	const float invRing = 1.f - ring;
+	const bool parallelFB = (routeMode >= 1);
+	const bool routeAdd = (routeMode == 1);
+
+	const bool doHarm = (harmonics > 0.f);
+	const float harmFactor = harmonics * 20.f + 1.f;
+	const float harmDiv = 1.f / (harmonics * 0.3f + 1.f);
+
+	const float dampCoeff = std::exp(-2.f * std::numbers::pi_v<float> * delayDamp / m_sampleRate);
+	const float toneCoeff = std::exp(-2.f * std::numbers::pi_v<float> * tone / m_sampleRate);
```

Comment:
You've already calculated `twoPi` (at compile-time as well, ideally) so there's no need to do it twice more:

```suggestion
	const float dampCoeff = std::exp(-twoPi * delayDamp / m_sampleRate);
	const float toneCoeff = std::exp(-twoPi * tone / m_sampleRate);
```